pkg/apis/core/validation/namespacedcloudprofile.go

```diff
@@ -0,0 +1,97 @@
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&cloudProfile.ObjectMeta, true, ValidateName, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateNamespacedCloudProfileSpec(&cloudProfile.Spec, field.NewPath("spec"))...)
+
+	return allErrs
+}
+
+// ValidateNamespacedCloudProfileUpdate validates a CloudProfile object before an update.
+func ValidateNamespacedCloudProfileUpdate(newProfile, oldProfile *core.NamespacedCloudProfile) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newProfile.ObjectMeta, &oldProfile.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateNamespacedCloudProfileSpecUpdate(&newProfile.Spec, &oldProfile.Spec, field.NewPath("spec"))...)
+	allErrs = append(allErrs, ValidateNamespacedCloudProfile(newProfile)...)
+
+	return allErrs
+}
+
+// ValidateNamespacedCloudProfileSpecUpdate validates the spec update of a CloudProfile.
+func ValidateNamespacedCloudProfileSpecUpdate(oldProfile, newProfile *core.NamespacedCloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(oldProfile.Parent, newProfile.Parent, fldPath.Child("parent"))...)
+
+	return allErrs
+}
+
+// ValidateNamespacedCloudProfileSpec validates the specification of a NamespacedCloudProfile object.
+func ValidateNamespacedCloudProfileSpec(spec *core.NamespacedCloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, validateParent(spec.Parent, fldPath.Child("parent"))...)
+
+	if spec.Kubernetes != nil {
+		allErrs = append(allErrs, validateKubernetesSettings(*spec.Kubernetes, fldPath.Child("kubernetes"))...)
+	}
+	if spec.MachineTypes != nil {
+		allErrs = append(allErrs, validateMachineImages(spec.MachineImages, fldPath.Child("machineImages"))...)
+	}
+	if spec.MachineTypes != nil {
+		allErrs = append(allErrs, validateMachineTypes(spec.MachineTypes, fldPath.Child("machineTypes"))...)
+	}
+	if spec.VolumeTypes != nil {
+		allErrs = append(allErrs, validateVolumeTypes(spec.VolumeTypes, fldPath.Child("volumeTypes"))...)
+	}
+	if spec.Regions != nil {
+		allErrs = append(allErrs, validateRegions(spec.Regions, fldPath.Child("regions"))...)
+	}
+	if spec.CABundle != nil {
+		_, err := utils.DecodeCertificate([]byte(*(spec.CABundle)))
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("caBundle"), *(spec.CABundle), "caBundle is not a valid PEM-encoded certificate"))
+		}
+	}
+
+	return allErrs
+}
+
+func validateParent(parent core.CloudProfileReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if parent.Kind != "CloudProfile" {
+		allErrs = append(allErrs, field.NotSupported(fldPath.Child("kind"), parent.Kind, []string{"NamespacedCloudProfile"}))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.NotSupported(fldPath.Child("kind"), parent.Kind, []string{"CloudProfile"}))
```
Please enhance the test with detail of the error as well, and if possible, include the Detail field for other tests as well.

Reply:
Enhanced it for the tests in https://github.com/gardener/gardener/pull/9440/commits/026d5bdc62ce06a524e6839f6d663a7dcf36a6e6. The rest of the tests were copied over from the CloudProfile.

Reply:
Yes I agree that most of the existing tests contain the "Detail" field, but it would be great if the new tests would add this, to prevent missing bugs like this.

Reply:
Is it really necessary to copy the tests here? Should we just unit test these functions once in the `cloudprofile_test.go` itself by using the `validation` package instead of `validation_test`?

Reply:
Probably, however since all of the tests use the `ValidateNamespacedCloudProfile` or `ValidateNamespacedCloudProfileUpdate` method, I think the "duplication" is okay here since there are no actual duplicate tests.